websites/S/Serial Podcast/presence.ts

```diff
@@ -0,0 +1,340 @@
+
+async function getShortURL(url: string) {
+	if (!url || url.length < 256) return url;
+	if (shortenedURLs[url]) return shortenedURLs[url];
+	try {
+		const pdURL = await (
+			await fetch(`https://pd.premid.app/create/${url}`)
+		).text();
+		shortenedURLs[url] = pdURL;
+		return pdURL;
+	} catch (err) {
+		presence.error(err);
+		return url;
+	}
+}
+
+function capitaliseFirstLetter(string: string) {
+	return (
+		string?.trim().charAt(0).toUpperCase() +
+		string?.trim().slice(1).toLowerCase()
+	);
+}
+
+async function getStrings() {
+	return presence.getStrings(
+		{
+			viewHome: "general.viewHome",
+			browse: "general.browsing",
+			play: "general.playing",
+			pause: "general.paused",
+			buttonViewEpisode: "general.buttonViewEpisode",
+			buttonViewPage: "general.buttonViewPage",
+			viewEpisode: "general.viewEpisode",
+			searchFor: "general.searchFor",
+			listeningTo: "general.listeningTo",
+			episode: "general.episode",
+			season: "general.season",
+			viewing: "general.viewing",
+			readingPost: "general.readingPost",
+			buttonListenAlong: "general.buttonListenAlong",
+			reading: "general.reading",
+		},
+		await presence.getSetting<string>("lang").catch(() => "en")
+	);
+}
+
+enum Assets {
+	Logo = "https://i.imgur.com/z271J2A.jpg",
+	Pause = "https://i.imgur.com/NyZsbVO.png",
+	Play = "https://i.imgur.com/Y1m0KVP.png",
+	Searching = "https://i.imgur.com/oGQtnIY.png",
+	Reading = "https://i.imgur.com/8vMPNni.png",
+}
+
+let strings: Awaited<ReturnType<typeof getStrings>>,
+	oldLang: string = null;
+
+presence.on("UpdateData", async () => {
+	const { href, pathname } = document.location,
+		[showTimestamp, showButtons, newLang, showCovers, privacy] =
+			await Promise.all([
+				presence.getSetting<boolean>("timestamp"),
+				presence.getSetting<boolean>("buttons"),
+				presence.getSetting<string>("lang").catch(() => "en"),
+				presence.getSetting<boolean>("covers"),
+				presence.getSetting<boolean>("privacy"),
+			]),
+		podcast: HTMLAudioElement = document.querySelector("audio#jp_audio_0"),
+		otherInfoPages =
+			document.querySelector("div.node.node-page > div.inner > h1")
+				?.textContent ??
+			document.querySelector("header.node-header > h1")?.textContent;
+
+	if (oldLang !== newLang || !strings) {
+		oldLang = newLang;
+		strings = await getStrings();
+	}
+
+	let presenceData: PresenceData = {
+		largeImageKey: Assets.Logo,
+		details: "Unsupported page",
+		smallImageKey: Assets.Reading,
+		smallImageText: strings.reading,
+	};
+
+	if (privacy) {
+		presenceData.details = strings.browse;
+		presence.setActivity(presenceData);
+		return;
+	}
+
+	const pages: Record<string, PresenceData> = {
+		"/about": {
+			details: strings.viewing,
+			state: "About Serial Podcast",
+			buttons: [
+				{
+					label: strings.buttonViewPage,
+					url: href,
+				},
+			],
+		},
+	};
+
+	for (const [path, data] of Object.entries(pages))
+		if (pathname.includes(path)) presenceData = { ...presenceData, ...data };
```

Comment:
only 1 in there, just make it an if case

Reply:
Will fix soon 👍